OrbitGl/CaptureSerializer.cpp

```diff
@@ -26,87 +30,107 @@
 #include "absl/strings/str_format.h"
 #include "capture_data.pb.h"
 
+static absl::flat_hash_map<const uint32_t, const std::string>
+    capture_version_to_orbit_version_map{
+        {1, "1.50"},
+    };
+
 //-----------------------------------------------------------------------------
 ErrorMessageOr<void> CaptureSerializer::Save(const std::string& filename) {
   Capture::PreSave();
 
+  header.set_version(kCurrentCaptureVersion);
+
   std::ofstream file(filename, std::ios::binary);
   if (file.fail()) {
     ERROR("Saving capture in \"%s\": %s", filename, "file.fail()");
     return ErrorMessage("Error opening the file for writing");
   }
 
-  try {
+  {
     SCOPE_TIMER_LOG(absl::StrFormat("Saving capture in \"%s\"", filename));
     Save(file);
-  } catch (std::exception& e) {
-    ERROR("Saving capture in \"%s\": %s", filename, e.what());
-    return ErrorMessage("Error serializing the capture");
   }
 
   return outcome::success();
 }
 
 void CaptureSerializer::Save(std::ostream& stream) {
-  cereal::BinaryOutputArchive archive(stream);
   std::basic_ostream<char> Stream(&GStreamCounter);
-  cereal::BinaryOutputArchive CountingArchive(Stream);
+  google::protobuf::io::OstreamOutputStream out_counting_stream(&Stream);
+  google::protobuf::io::CodedOutputStream coded_counting_output(
+      &out_counting_stream);
+
   GStreamCounter.Reset();
-  SaveImpl(CountingArchive);
+  SaveImpl(&coded_counting_output);
   PRINT_VAR(GStreamCounter.Size());
   GStreamCounter.Reset();
```

Comment:
I know it has always been there, but honestly do we even need this part? This serializes just to print the size? It's probably quite an unnecessary overhead.

Reply:
I don't know if somebody uses it. But it looks like I already removed the useful part and now it shows only the total  size which is not so interesting. Btw, if we need the size we always can just change WriteMessage and log message_size.